Build failure with GHC 9.12 (base-4.21)

I think for supporting GHC 9.12 a new release is needed.
```
$ cabal build -w ghc-9.12.0 --allow-newer=base
...
orphans/Generic/Data/Orphans.hs:17:10: error: [GHC-59692]
    Duplicate instance declarations:
      instance Eq1 V1 -- Defined at orphans/Generic/Data/Orphans.hs:17:10
      instance [safe] Eq1 V1 -- Defined in ‘Data.Functor.Classes’
   |
17 | instance Eq1 V1 where
   |          ^^^^^^
...
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,8 +1,14 @@
+# 1.1.0.1
+
+- Compatibility with *base* 4.21 (GHC 9.12)
+- Avoid dependency on *contravariant*
+
 # 1.1.0.0
 
 - Remove `Eq` and `Ord` for `Generically1` instances.
   They are now available in *base* 4.18, and *base-orphans* 0.8.8
   for backwards compatibility.
+- (Revisions) Also compatible with *base* 4.20 (GHC 9.10) and 4.19 (GHC 9.8)
 
 # 1.0.0.1
 

--- a/generic-data.cabal
+++ b/generic-data.cabal
@@ -46,7 +46,7 @@ library
     base-orphans >= 0.8.8,
     ghc-boot-th,
     show-combinators,
-    base >= 4.9 && < 4.21
+    base >= 4.9 && < 4.22
   if impl(ghc < 9.6)
     build-depends: contravariant
   hs-source-dirs:      orphans

--- a/generic-data.cabal
+++ b/generic-data.cabal
@@ -1,5 +1,5 @@
 name:                generic-data
-version:             1.1.0.0
+version:             1.1.0.1
 synopsis:            Deriving instances with GHC.Generics and related utilities
 description:
   Generic implementations of standard type classes.

--- a/orphans/Generic/Data/Orphans.hs
+++ b/orphans/Generic/Data/Orphans.hs
@@ -1,14 +1,19 @@
 {-# OPTIONS_GHC -Wno-orphans #-}
 
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE EmptyCase #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE TypeOperators #-}
 
--- | Orphan instances. They should probably be upstreamed.
+-- | Orphan instances.
+--
+-- The orphan instances in this module have been upstreamed in base 4.21 (GHC 9.12).
+-- This module is empty starting from that version. It remains for backwards compatiblity.
 
 module Generic.Data.Orphans where
 
+#if __GLASGOW_HASKELL__ < 912
 import Data.Functor.Classes
 import Data.Orphans ()
 import Data.Semigroup
@@ -71,3 +76,4 @@ instance (Eq1 f, Eq1 g) => Eq1 (f :.: g) where
 instance (Ord1 f, Ord1 g) => Ord1 (f :.: g) where
   liftCompare compare' (Comp1 x1) (Comp1 x2) =
     (liftCompare . liftCompare) compare' x1 x2
+#endif

